### PR TITLE
Restructure Code

### PR DESCRIPTION
**Pull request by perrymanmd**
Restructured code
* Moved main script to `shef` packaged and renamed (`shefParser` -> `shef.shef_parser.py`)
  * Recreated `shefParser` as a thin wrapper that just calls `shef.shef_parser.main()`
* Moved `shef_loader` package to `shef.loaders` subpackage
* Added `parse()` function to main script
  * Handles everything but command line parsing and SHEFPARM generation
  * Called by `main()`
  * Can be called from other scripts to bypass command line
Improved use, handling, and logging of exceptions
Corrected `shef.loaders.shared.SHEF_ENGLISH_UNITS` entry for PE code `PL` to 'mb' (manual has 'in-hg')

### CODE REVIEW

File: shef/shef_parser.py [hunk 1]
```diff
@@ -0,0 +1,3858 @@
...
+        log_name: Optional[str],
+        log_level: str,
+        append_log: bool,
+        log_timestamps: bool,
+        shefparm: Optional[str],
+        use_defaults: bool,
+        shefit_times: bool,
+        reject_problematic: bool,
+        loader_spec: str,
+        unload: bool,) :
```
Comment (jbkolze) [inline shef/shef_parser.py:3477]: Could we provide defaults for some of these?  Entering values for all 14 arguments is pretty tedious when you a user be fine with the defaults for several/most of them.

File: shef/shef_parser.py
Comment (jbkolze) [inline shef/shef_parser.py:1]: Seems like the error handling has changed with this revision.  Previously, most errors were handled and logged by the parser.  For example, if a SHEF code was not recognized by the loader the following would be logged:
```
ERROR: 'WNSI3L.QRQFUZ' in message starting at example-shef-
        files/lrl/ohrfc_LRL_0hrQPF_Locals_2024070212.2024.07.02.124124:2594
INFO: Error is in message :
        .ER WNSI3L 20240627 Z DH1200/DC202407021238/QRIFUZ /DIH06
        .E1  0.085/0.1/0.107/0.103/0.093/0.08/0.067/0.057/
        .E2  0.05/0.045/0.041/0.037/0.033/0.03/0.028/0.025/
        .E3  0.023/0.021/0.019/0.017/0.015/0.014/0.012/0.01/
        .E4  0.009/0.008/0.007/0.006/0.005/0.004/0.004/0.003/
        .E5  0.002/0.002/0.001/0.001/0.001/0.001/0.001/0/
        .E6  0/0/0/0/0/0/0/0/
        .E7  0/0/0/0/0/0/0/0/
        .E8  0/0/0/0/0/
```
Now these errors are rising all the way to the top, ending execution, and throwing a traceback:
```
Traceback (most recent call last):
  File "/home/bkolze/projects/SHEF_processing/test.py", line 3, in <module>
    shef.shef_parser.parse(
  File "/home/bkolze/projects/SHEF_processing/shef/shef_parser.py", line 3637, in parse
    loader.set_shef_value(format_1_str)
  File "/home/bkolze/projects/SHEF_processing/shef/loaders/base_loader.py", line 103, in set_shef_value
    if self.get_time_series_name(shef_value) == self.time_series_name :
  File "/home/bkolze/projects/SHEF_processing/shef/loaders/cda_loader.py", line 170, in get_time_series_name
    return self._transforms[transform_key].timeseries_id
KeyError: 'NPTO1.QRQFUZ'
```
The traceback is thrown both when called from the command line and when called from within python.

**Review by jbkolze — CHANGES_REQUESTED** [review]
Parse function defaults aren't too big of a deal, but the error-handling issue seems like it might need to be resolved before merging this.  As it stands now, every SHEF code in a file has to be handled/mapped or the parser can't execute successfully.

File: shef/shef_parser.py (hunk 1 above)
Comment (perrymanmd) [reply shef/shef_parser.py:3477]: Sure.

File: shef/shef_parser.py
Comment (perrymanmd) [reply shef/shef_parser.py:1]: I did change some stuff, but I didn't mean for run-of-the mill parse errors to propagate out. Will correct.

Comment (perrymanmd) [reply shef/shef_parser.py:1]: I submitted a new commit to the PR branch.

**Review by jbkolze — APPROVED** [review]
Looks good.  Tested successfully on my end.  Thanks!

**Pull request merged**